APISpecification.yaml

```diff
@@ -0,0 +1,82 @@
+openapi: 3.0.3
+info:
+  title: Ensembl Resolver API
+  description: Resolver API resolves external urls to Ensembl
+  contact:
+    email: ensembl-webteam@ebi.ac.uk
+  version: 0.0.1
+servers:
+  - url: https://resolver.ensembl.org
+tags:
+  - name: resolver
+    description: Resolver API resolves external urls to Ensembl
+paths:
+  /resolve/{gene_stable_id}:
+    get:
+      summary: Resolve stable ID with optional query params
+      description: Resolves to a beta url when a stable id wih optional query params provided
+      parameters:
+        - name: gene_stable_id
+          in: path
+          description: Gene stable id used to represent a gene in a genome.
+          required: true
+          explode: true
+          schema:
+            type: string
+            example: ENSG00000139618
+        - name: type
+          in: query
+          description: Defines a type - gene/transcript/variant etc
+          required: false
+          schema:
+            type: string
+        - name: app
+          in: query
+          description: Choose between apps - Genome Browser/ Entity Viewer
+          required: false
+          schema:
+            type: string
+        - name: gca
+          in: query
+          description: GCA assembly accession id of the genome
+          required: false
+          schema:
+            type: string
+       responses:
+        '302':
```

Comment:
Here is where I disagree with Andy — he thinks it's a 302 (moved temporarily), whereas I think it should be a 301 (moved permanently). For web crawlers of search engines, a 301 is better, because it is a stronger suggestion for indexing the resolved url.

Reply:
Hi @andrewyatz any comment on this?

Reply:
I believe he thinks 302 is better because the underline url can change so if the crawlers records the resolved url they will be rendered invalid when a breaking change is made

Reply:
> if the crawlers records the resolved url they will be rendered invalid when a breaking change is made

There is an internal contradiction in this.

- **Proposition 1:** We do want Ensembl pages to be in Google search index, right? (Whether we want them to index pages while the site is in beta is a separate question; but we do want Ensembl urls to get into search engines' databases, don't we?)
- **Proposition 2:** As our website evolves, there is always a risk that some urls will eventually have to change — this is something we cannot foresee from where we are now. But this still does not invalidate proposition 1, right?

So, why do we worry about changed urls specifically in the context of the resolver service? If a search engine indexes a url and the url changes, it does not matter whether search engine got that url from a 301 redirect or from a direct navigation to that web page.